plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergInternalRetry.java

```diff
@@ -88,6 +90,16 @@ public synchronized void createTable(Table table, PrincipalPrivileges principalP
                 super.createTable(table, principalPrivileges);
                 throw new TableAlreadyExistsException(table.getSchemaTableName());
             }
+
+            @Override
+            public synchronized void dropTable(String databaseName, String tableName, boolean deleteData)
+            {
+                // Simulate retry mechanism with timeout failure of ThriftHiveMetastore.
+                // 1. dropTable correctly drops table but timeout is triggered
+                // 2. Retry to dropTable throws TableNotFoundException
+                super.dropTable(databaseName, tableName, deleteData);
+                throw new TableNotFoundException(schemaTableName(databaseName, tableName));
```

Comment:
nit/optional: we should do this only for tables named `test_drop_internal_retry`

Reply:
By leaving it as it we also test if dropping tables during unregister procedure.